### PR TITLE
HARVESTER: storage class polish

### PR DESCRIPTION
**Pull request by DaiYuzeng**
#### PR Checklist
- ~Is this a multi-tenancy feature/bug?~
    - [ ] ~Yes, the relevant RBAC changes are at:~
- ~Do we need to backport changes to the [old Rancher UI](https://github.com/rancher/u), such as RKE1?~
    - [ ] ~Yes, the relevant PR is at:~
- ~Are backend engineers aware of UI changes?~
    - [ ] ~Yes, the backend owner is:~

### Occurred changes and/or fixed issues

- https://github.com/harvester/harvester/issues/3686

### Areas or cases that should be tested

1. Go to manage Storage Class
2. Switch to Allowed Topologies tab
3. You can manage the Allowed Topologies

### CODE REVIEW

**Comment by DaiYuzeng** [comment]
image

Storage Class can only be created, can not be modified.

File: pkg/harvester/edit/harvesterhci.io.storage/index.vue [hunk 1]
```diff
@@ -128,6 +142,16 @@ export default {
...
+
+    allowedTopologies(neu) {
+      if (!neu || neu.length === 0) {
+        delete this.value.allowedTopologies;
+
+        return;
+      }
+
+      this.value.allowedTopologies = [{ matchLabelExpressions: neu }];
     }
```
Comment (WuJun2016) [inline pkg/harvester/edit/harvesterhci.io.storage/index.vue:155]: image

A small enhancement: When entering a value for "allowedTopologies" and then deleting it, there may be an issue when saving.

**Review by WuJun2016 — COMMENTED** [review]
other LGTM.

File: pkg/harvester/edit/harvesterhci.io.storage/index.vue (hunk 1 above)
Comment (DaiYuzeng) [reply pkg/harvester/edit/harvesterhci.io.storage/index.vue:155]: done

**Comment by DaiYuzeng** [comment]
PTAL, thanks! @WuJun2016

**Review by WuJun2016 — APPROVED** [review]
LGTM, thanks

**Pull request merged**